korman/properties/modifiers/sound.py

```diff
@@ -22,9 +22,133 @@
 
 from ... import korlib
 from .base import PlasmaModifierProperties
+from .physics import surface_types
 from ...exporter import ExportError
 from ... import idprops
 
+_randomsound_modes = {
+    "normal": plRandomSoundMod.kNormal,
+    "norepeat": plRandomSoundMod.kNoRepeats,
+    "coverall": plRandomSoundMod.kCoverall | plRandomSoundMod.kNoRepeats,
+    "sequential": plRandomSoundMod.kSequential
+}
+
+class PlasmaRandomSound(PlasmaModifierProperties):
+    pl_id = "random_sound"
+    pl_depends = {"soundemit"}
+
+    bl_category = "Logic"
+    bl_label = "Random Sound"
+    bl_description = ""
+
+    mode = EnumProperty(name="Mode",
+                        description="Playback Type",
+                        items=[("random", "Random Time", "Plays a random sound from the emitter at a random time"),
+                               ("collision", "Collision Surface", "Plays a random sound when the object's parent collides")],
+                        default="random",
+                        options=set())
+
+    # Physical (read: collision) sounds
+    play_on = EnumProperty(name="Play On",
+                           description="Play sounds on this collision event",
+                           items=[("slide", "Slide", "Plays a random sound on object slide"),
+                                  ("impact", "Impact", "Plays a random sound on object slide")],
+                           options=set())
+    surfaces = EnumProperty(name="Play Against",
+                            description="Sounds are played on collision against these surfaces",
+                            items=surface_types[1:],
+                            options={"ENUM_FLAG"})
+
+    # Timed random sounds
+    auto_start = BoolProperty(name="Auto Start",
+                              description="Start playing when the Age loads",
+                              default=True,
+                              options=set())
+    play_mode = EnumProperty(name="Play Mode",
+                             description="",
+                             items=[("normal", "Any", "Plays any attached sound"),
```

Comment:
Would it be too much/too confusing to add the mode numbers after the Mode names (e.g. "Any (16)", "No Repeat (17)", "Full Set (19)", "Sequential (48)"? It could be helpful for anyone converting Cyan Ages and looking at the PRPs.

If it's too much, we could just leave it to future wiki documentation.

Reply:
I think it might be better to add support for RSMs to ZLZ

Reply:
That'd work too.

Reply:
> add support for RSMs to ZLZ

Sure, this (and a bunch of other things) are planned for when I jump back to ZLZ. I'm always interested in supporting more Korman features. Having the user deal with the underlying numbers 'n stuff is always undesirable.

I was planning on doing some other stuff before that, but I can prioritize this if you need it Doobes.

Reply:
@Jrius If it's not too difficult, of course. It'd be one less thing to have to wire when converting.

Reply:
'K, support added to ZLZ. This also required the object's collider to parent all other objects (drawable, sounds etc), which is better behavior anyway.

Observation: Relto firemarbles use collision sound group kUser1 (great naming...), which is not available to Korman and therefore ignored. Dunno if you want to support it.

Reply:
Oh... of course Cyan used those groups. 🙄